cumulusci/tasks/salesforce/tests/test_base_tasks.py

```diff
@@ -35,7 +35,11 @@ def test_run_task(self):
                 task()
 
     def test_update_credentials(self):
-        def update_config(keychain):
+        update_config_called = False
+
+        def update_config(keychain, save=False):
+            nonlocal update_config_called
+            update_config_called = True
             self.org_config.config["new"] = "new"
 
         self.org_config.refresh_oauth_token = update_config
```

Comment:
You could potentially use a mock with `side_effect` here to update the config without needing your own logic for keeping track of whether it was called.

Reply:
I would probably have done that if I could do the assignment in a lambda, but since I still need to write a full function for the assignment, I think the number of lines of code would be similar.

Reply:
I started to make that change to see how it would work out, then realized there's a more significant problem: the test used to verify that running the task saved the org config in the keychain, but now it only asserts that refresh_oauth_token was called.

Reply:
Okay I added back that assertion: 546545b5